#! /bin/sh /usr/share/dpatch/dpatch-run
## f-spot-import-gphoto.dpatch by Martin Pitt <martin.pitt@ubuntu.com>
##
## DP: Fix f-spot-import to understand GIO gphoto:// URLs. Also speculatively
## DP: gvfs-umount the camera, otherwise F-Spot cannot communicate with it.
## DP: (Gnome #551803)
@DPATCH@
diff -urNad f-spot-0.5.0.2~/tools/f-spot-import f-spot-0.5.0.2/tools/f-spot-import
--- f-spot-0.5.0.2~/tools/f-spot-import	2008-07-08 13:01:43.000000000 +0200
+++ f-spot-0.5.0.2/tools/f-spot-import	2008-10-16 10:05:56.000000000 +0200
@@ -1,6 +1,21 @@
 #!/bin/bash
 
 udi="$1"
+
+if [ "$udi" != "${udi#gphoto2:}" ]; then
+    # gphoto gvfs source, pass it on unmodified
+
+    # make sure it's not mounted by gvfs, otherwise we can't access the camera
+    # (current F-Spot version does not yet understand GIO)
+    gvfs-mount -u "$udi" || true
+
+    # transform GIO URL to a format f-spot understands
+    uri=`echo "$udi" | sed 's_://\[_:_; s_/$__; s_\]$__'`
+
+    exec f-spot --import "$uri"
+    exit 1
+fi
+
 #xmessage $udi
 mount_point=`hal-get-property --udi="$udi" --key=volume.mount_point`
 if [ -n "$mount_point" ]; then
